File: QE-Selenium-Python/Activity8.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

driver = webdriver.Firefox()
wait = WebDriverWait(driver, 10)
actions = ActionChains(driver)

driver.get("https://training-support.net/webelements/mouse-events")
print("Title:", driver.title)

cargo_lock = driver.find_element(By.CSS_SELECTOR, ".file:nth-child(1)")
cargo_toml = driver.find_element(By.CSS_SELECTOR, ".file:nth-child(2)")
src = driver.find_element(By.CSS_SELECTOR, ".file:nth-child(3)")
target = driver.find_element(By.CSS_SELECTOR, ".file:nth-child(4)")

# Click Cargo.lock → move → click Cargo.toml
actions.click(cargo_lock).move_to_element(cargo_toml).click().perform()
print(driver.find_element(By.ID, "result").text)

# Double click src → right click target → Open
actions.double_click(src).context_click(target).perform()

open_option = wait.until(
    EC.element_to_be_clickable((By.CSS_SELECTOR, "li.w-full:nth-child(1)"))
)
open_option.click()

print(driver.find_element(By.ID, "result").text)

driver.quit()

File: QE-Selenium-Java/Activity8.java
package demos;

import java.time.Duration;
import org.openqa.selenium.*;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.*;

public class Activity8 {
    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();
        Actions actions = new Actions(driver);
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));

        driver.get("https://training-support.net/webelements/mouse-events");
        System.out.println("Title: " + driver.getTitle());

        WebElement cargoLock = driver.findElement(By.cssSelector(".file:nth-child(1)"));
        WebElement cargoToml = driver.findElement(By.cssSelector(".file:nth-child(2)"));
        WebElement src = driver.findElement(By.cssSelector(".file:nth-child(3)"));
        WebElement target = driver.findElement(By.cssSelector(".file:nth-child(4)"));

        // Click Cargo.lock → move to Cargo.toml → click
        actions.click(cargoLock).moveToElement(cargoToml).click().perform();
        System.out.println(driver.findElement(By.id("result")).getText());

        // Double click src → right click target → Open
        actions.doubleClick(src).contextClick(target).perform();

        WebElement open = wait.until(
            ExpectedConditions.elementToBeClickable(
                By.cssSelector("li.w-full:nth-child(1)")
            )
        );
        open.click();

        System.out.println(driver.findElement(By.id("result")).getText());

        driver.quit();
    }
}